pkg/services/alerting/extractor.go

```diff
@@ -203,16 +228,53 @@ func (e *DashAlertExtractor) getAlertFromPanels(jsonWithPanels *simplejson.Json,
 	return alerts, nil
 }
 
-func validateAlertRule(alert *models.Alert) bool {
-	return alert.ValidToSave()
+func validateAlertRule(alert *models.Alert) (ok bool, reason string) {
+	ok = alert.ValidToSave()
+	if !ok {
+		reason = fmt.Sprintf("Dashboard ID, Org ID or Panel ID is not correct, alertName=%v, panelId=%v, orgId=%v, dashboardId=%v", alert.Name, alert.PanelId, alert.OrgId, alert.DashboardId)
+	}
+	return ok, reason
+}
```

Comment:
I don't see any point in having two return values, when we can instead just check if the returned reason is empty or not (it also reduces the possible bugs, since otherwise you could have `!ok` and an empty string):
```suggestion
func validateAlertRule(alert *models.Alert) string {
	if alert.ValidToSave() {
		return ""
	}

	return fmt.Sprintf("Dashboard ID, Org ID or Panel ID is not correct, alertName=%v, panelId=%v, orgId=%v, dashboardId=%v", alert.Name, alert.PanelId, alert.OrgId, alert.DashboardId)
}
```

Reply:
thanks! I have changed it to return an error; do you mind to have a fresh look?

Reply:
@papagian Have you pushed the change? I see the same code as before.

Reply:
yes, that's the commit: ad933863e32057cf02caabaa5799e3c14033770d